src/lib/FlightTasks/tasks/Manual/FlightTaskManual.hpp

```diff
@@ -59,6 +59,7 @@ class FlightTaskManual : public FlightTask
 	bool _sticks_data_required = true; /**< let inherited task-class define if it depends on stick data */
 	matrix::Vector<float, 4> _sticks; /**< unmodified manual stick inputs */
 	matrix::Vector<float, 4> _sticks_expo; /**< modified manual sticks using expo function*/
+	int _gear_switch_old = manual_control_setpoint_s::SWITCH_POS_NONE; /**< old switch state*/
```

Comment:
`_gear_switch_old = landing_gear_s::GEAR_KEEP `?

Reply:
No, to detect the transition of the RC **switch** controlling the landing gear (if there is any) like the word `switch` suggests.
It's states are defined according to: https://github.com/PX4/Firmware/blob/0c60fff6bd0e09db9b4fc716472849978a60516f/msg/manual_control_setpoint.msg#L3-L6